pkg/catalog/ingress.go

```diff
@@ -20,6 +20,24 @@ const (
 	singeIPPrefixLen = "/32"
 )
 
+// GetIngressTrafficPolicies returns a list of IngressTrafficPolicy objects for the given MeshService list
+func (mc *MeshCatalog) GetIngressTrafficPolicies(meshServices []service.MeshService) []*trafficpolicy.IngressTrafficPolicy {
+	var policies []*trafficpolicy.IngressTrafficPolicy
+
+	for _, meshSvc := range meshServices {
+		policy, err := mc.GetIngressTrafficPolicy(meshSvc)
+		if err != nil {
+			log.Error().Err(err).Msgf("Error getting ingress traffic policy for service %s, skippint it", meshSvc)
+			continue
+		}
+		if policy != nil {
```

Comment:
nit: 

would it make sense to change GetIngressTrafficPolicy to return an error if one doesn't exist? Even an error like 

```go
policy, err := GetIngressTrafficPolicy()
if err == errNotFound {
   continue // ignore
} else if err != nil {
 log(err)
continue
}

policies = append(...)
```

Small thing, but as a user, I'd expect if no err then not nil.

Reply:
Currently, it's okay for a policy to not exist, in which case we return `nil`. I left a TODO to remove GetIngressTrafficPolicy as the function even in its current form never returns an error. I'll consider handling the return semantics when I address that TODO to merge the 2 GetIngressPolic.* APIs. We simply log an error and ignore the erring policy, as we try to program as many configs on the proxy as possible.